### DISCUSSION TITLE
Debugers for language created with xtext

### DISCUSSION THREAD

**Discussion by mxtsuper** [eclipse-xtext/xtext#2749] [General]
Hello

What kind of debugers/solutions can be used to debug the language created with xtext, please?

Thanks.

**Comment by cdietrich** [comment]
in eclipse: when you use xbase it should work ootb. otherwise (when target still is java) https://blogs.itemis.com/en/debugging-dsls-in-xtext-and-eclipse

**Comment by mxtsuper** [comment]
Thanks cdietrich.  
What about the target is c/C++ ?
If I use the LSP, can I use the debuger ?

**Reply by cdietrich** [reply]
i dont know how c++ debugging works and if there is something like source maps for cpp and how they work.
but i assume you can use the trace information used in xbase and also in the blog post to generate source mappings or whatever c++ uses for debugging
